rpy2/rinterface_lib/embedded.py

```diff
@@ -232,10 +232,12 @@ def _setcallback(rlib, rlib_symbol: str,
 
 # TODO: can init_once() be used here ?
 def _initr(
-        interactive: bool = True,
+        interactive: typing.Optional[bool] = None,
         _want_setcallbacks: bool = True,
         _c_stack_limit: int = _DEFAULT_C_STACK_LIMIT
 ) -> typing.Optional[int]:
+    if interactive is None:
+        interactive = not os.environ.get('R_DISABLE_INTERACTIVE_MODE')
```

Comment:
I don't think this change persists as much as you'd hope to (see l. 295 in this file).